catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/PipelineRepository.java

```diff
@@ -82,10 +90,70 @@ public Pipeline setFields(Pipeline pipeline, Fields fields) throws IOException,
     if (!fields.contains("tasks")) {
       pipeline.withTasks(null);
     }
+    pipeline.setPipelineStatus(fields.contains("pipelineStatus") ? getPipelineStatus(pipeline) : null);
     pipeline.setTags(fields.contains("tags") ? getTags(pipeline.getFullyQualifiedName()) : null);
     return pipeline;
   }
 
+  private List<PipelineStatus> getPipelineStatus(Pipeline pipeline) throws IOException {
+    List<PipelineStatus> pipelineStatus =
+        JsonUtils.readObjects(
+            daoCollection.entityExtensionDAO().getExtension(pipeline.getId().toString(), "pipeline.pipelineStatus"),
+            PipelineStatus.class);
+    if (pipelineStatus != null) {
+      pipelineStatus.sort(
+          Comparator.comparing(s -> parseDate(s.getExecutionDate(), RestUtil.DATE_FORMAT), Comparator.reverseOrder()));
+    }
+    return pipelineStatus;
+  }
+
+  @Transaction
+  public Pipeline addPipelineStatus(UUID pipelineId, PipelineStatus pipelineStatus) throws IOException, ParseException {
+    // Validate the request content
+    Pipeline pipeline = daoCollection.pipelineDAO().findEntityById(pipelineId);
+    Map<String, PipelineStatus> storedMapStatus = new HashMap<>();
+
+    // Add stored status
+    List<PipelineStatus> storedPipelineStatus = getPipelineStatus(pipeline);
+    if (storedPipelineStatus != null) {
+      for (PipelineStatus status : storedPipelineStatus) {
+        storedMapStatus.put(status.getExecutionDate(), status);
+      }
+    }
+
+    // validate all the Tasks
+    for (Status taskStatus : pipelineStatus.getTaskStatus()) {
+      validateTask(pipeline, taskStatus.getName());
+    }
+
+    // Add new status
+    storedMapStatus.put(pipelineStatus.getExecutionDate(), pipelineStatus);
+
+    // Prepare to update status
+    List<PipelineStatus> updatedStatus = new ArrayList<>(storedMapStatus.values());
+
+    daoCollection
+        .entityExtensionDAO()
+        .insert(
+            pipelineId.toString(), "pipeline.pipelineStatus", "pipelineStatus", JsonUtils.pojoToJson(updatedStatus));
+    setFields(pipeline, Fields.EMPTY_FIELDS);
+    return pipeline.withPipelineStatus(getPipelineStatus(pipeline));
+  }
+
+  // Validate if a given task exists in the pipeline
+  private void validateTask(Pipeline pipeline, String taskName) {
+    boolean validTask = false;
```

Comment:
There is a Java lambda function called anyMatch which I believe does most of the logic that we have in this function